diff --git a/CMakeLists.txt b/CMakeLists.txt
index f74b1378f..a9aa7f3ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -218,7 +218,9 @@ option(BUILD_SHARED_LIBS "Create shared libraries by default" ON)
 include(${PROJECT_SOURCE_DIR}/cmake/common/eprosima_libraries.cmake)
 
 eprosima_find_package(fastcdr REQUIRED)
+set(EPROSIMA_BUILD ON)
 eprosima_find_thirdparty(Asio asio)
+set(EPROSIMA_BUILD OFF)
 eprosima_find_thirdparty(TinyXML2 tinyxml2)
 
 find_package(foonathan_memory REQUIRED)
diff --git a/thirdparty/asio b/thirdparty/asio
--- a/thirdparty/asio
+++ b/thirdparty/asio
@@ -1 +1 @@
-Subproject commit 22afb86087a77037cd296d27134756c9b0d2cb75
+Subproject commit 22afb86087a77037cd296d27134756c9b0d2cb75-dirty
diff --git a/thirdparty/fastcdr b/thirdparty/fastcdr
--- a/thirdparty/fastcdr
+++ b/thirdparty/fastcdr
@@ -1 +1 @@
-Subproject commit 53a0b8cae0b9083db69821be0edb97c944755591
+Subproject commit 53a0b8cae0b9083db69821be0edb97c944755591-dirty
